### PR TITLE
Update Golang veresion to 1.20.5

### PR DESCRIPTION
**Pull request by pkprzekwas**
**What this PR does / why we need it**:
Update Golang veresion to 1.20.5

**What type of PR is this?**
/kind chore

**Does this PR introduce a user-facing change? Then add your Release Note here**:
```release-note
Update to Go 1.20.5
```

**Documentation**:
```documentation
NONE
```

### CODE REVIEW

File: hack/images/web-terminal/Dockerfile [hunk 1]
```diff
@@ -16,10 +16,10 @@ FROM docker.io/alpine:3.17
 LABEL maintainer="support@kubermatic.com"
 
 # Source: https://dl.k8s.io/release/stable-1.25.txt
-ENV KUBECTL_VERSION=v1.25.8
+ENV KUBECTL_VERSION=v1.25.10
```
Comment (pkprzekwas) [inline hack/images/web-terminal/Dockerfile:19]: Do we want to upgrade to 1.27.x?

Comment (ahmedwaleedmalik) [reply hack/images/web-terminal/Dockerfile:19]: v1.26.x would be more appropriate

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Pull request merged**